tests/format/typescript/assignment/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -180,6 +180,116 @@ const firestorePersonallyIdentifiablePaths: Array<Collections.Users.Entity> =
 ================================================================================
 `;
 
+exports[`issue-10848.tsx format 1`] = `
+====================================options=====================================
+parsers: ["typescript"]
+printWidth: 80
+                                                                                | printWidth
+=====================================input======================================
+interface MyComponentProps {
+  x: number
+}
+
+const MyComponent: React.VoidFunctonComponent<MyComponentProps> = ({ x }) => {
+  const a = useA()
+  return <div>x = {x}; a = {a}</div>
+}
+
+interface MyComponent2Props {
+  x: number
+  y: number
+}
+
+const MyComponent2: React.VoidFunctonComponent<MyComponent2Props> = ({ x, y }) => {
+  const a = useA()
+  return <div>x = {x}; y = {y}; a = {a}</div>
+}
+
+interface MyComponentWithLongNameProps {
+  x: number
+  y: number
+  [key: string]: unknown
+}
+
+const MyComponentWithLongName: React.VoidFunctonComponent<MyComponentWithLongNameProps> = ({ x, y }) => {
+  const a = useA()
+  return <div>x = {x}; y = {y}; a = {a}</div>
+}
+
+const MyComponentWithLongName: React.VoidFunctonComponent<MyComponentWithLongNameProps> = ({ x, y, anotherPropWithLongName1, anotherPropWithLongName2, anotherPropWithLongName3, anotherPropWithLongName4 }) => {
+  const a = useA()
+  return <div>x = {x}; y = {y}; a = {a}</div>
+}
+
+=====================================output=====================================
+interface MyComponentProps {
+  x: number;
+}
+
+const MyComponent: React.VoidFunctonComponent<MyComponentProps> = ({ x }) => {
+  const a = useA();
+  return (
+    <div>
+      x = {x}; a = {a}
+    </div>
+  );
+};
+
+interface MyComponent2Props {
+  x: number;
+  y: number;
+}
+
+const MyComponent2: React.VoidFunctonComponent<MyComponent2Props> = ({
+  x,
+  y,
+}) => {
```

Comment:
This shouldn't happen

Reply:
Are you suggesting that we should see this?

```
const MyComponent2: (
  React.VoidFunctonComponent<MyComponent2Props>
) = ({ x, y }) => {





    
    
      
      x,




    
    
      
      y,




    
    
      
    <span class="blob-code-inner blob-code-marker" data-code-marker="+">}) </span>

Reply:
> Are you suggesting that we should see this?

Yes

> What about the fourth case

It's okay. Breaking the signature is unavoidable in that case.

Reply:
Thanks for the feedback @thorn0! Do the snapshots make sense to you know? If yes, happy to look into the implementation. Any hints & tips are welcome!

Reply:
Yes, looks good. See https://github.com/prettier/prettier/issues/10848#issuecomment-849607441